### ISSUE TITLE
Field Class in Feature Field Hgd is missing a Field

### ISSUE THREAD

**Issue by PFRYCAC** [hougaard/Simple-Object-Designer#257]
I'm a new to SOD user trying to set up a few new tables with relations.

I've had a few error messages and finding them not really easy to decipher. Though I was able to solve the first couple, I can't figure this one out when trying to publish:

"Field Class in Feature Field Hgd is missing a Field for the filter on , cannot publish."

I removed all of my Lookups and Filters one by one to see which Filter was the issue, and now they're all removed but the error persists. 

Thanks,
Paul

**Comment by hougaard** [comment]
Is there a "Copy Details" on that error ... to get some details?

**Comment by hougaard** [comment]
I'm pretty sure there's still a filter line sitting there with a field no = 0 (can sometime look like an empty/blank line)

**Comment by PFRYCAC** [comment]
Hi Erik

Sorry, I though that would only copy the message I was viewing!

Here's the paste:



Field Class in Feature Field Hgd is missing a Field for the filter on , cannot publish.

Internal session ID: 
0591f3d2-f365-4340-b6f3-5171d4cb0f5e

Application Insights session ID: 
8976ac72-b179-43ef-a01b-73277cf7e990

Client activity id: 
325c8af8-f867-4919-936b-b13b1d4b72a3

Time stamp on error: 
2023-08-03T06:04:16.4900180Z

AL call stack: 
"Build Feature Code Hgd"(CodeUnit 70310254).ExportTableFields line 174 - Simple Object Designer by Hougaard
"Build Feature Code Hgd"(CodeUnit 70310254).ExportTable line 30 - Simple Object Designer by Hougaard
"Build Extension Hgd"(CodeUnit 70310250).CreateExtensionEx line 290 - Simple Object Designer by Hougaard
"Object Designer Menu Hgd"(Page 70310263)."ExportExtension - OnAction"(Trigger) line 6 - Simple Object Designer by Hougaard

**Comment by PFRYCAC** [comment]
> I'm pretty sure there's still a filter line sitting there with a field no = 0 (can sometime look like an empty/blank line)

I've made 4 features and rechecked each - I don't see any field No 0 or any more than a single blank line at the bottom of the field list

**Comment by hougaard** [comment]
The stack trace confirms, there's a blank filter line on the "Class" field.

**Comment by PFRYCAC** [comment]
This is the only table I have with a field called Class...

![image](https://github.com/hougaard/Simple-Object-Designer/assets/136421714/1fe62a46-0117-48b8-b58f-2cab2d7e5b4a)

![image](https://github.com/hougaard/Simple-Object-Designer/assets/136421714/421ddfe7-dda0-477d-8dcb-a15a39892ff5)

**Comment by PFRYCAC** [comment]
I deleted and re-added the field, and the error has moved on to the next Filter I had previously. The answer looks like it will be to delete these fields and re-add.

**Comment by hougaard** [comment]
.. did you change field types? I wonder if you had lookup fields (or flowfield) and then changed them into normal fields... There might be "leftovers" in that case?

**Comment by PFRYCAC** [comment]
Yes, that's exactly what I did once I saw the first error. So I think if you change the field type without first deleting the line on the Filter, it will remain there and cause the error. 

Can you point me to a video which describes using the Lookup/Filters in the most detail? Thank you

(User generated) errors aside, this application is great and will definitely be taking it up.

**Comment by hougaard** [comment]
Yeah, I probably need to handle that case better....

I don't think I have a video about lookup field filters, but it's kinda the same as with flowfields:
https://www.youtube.com/playlist?list=PLnldREWlGR0twSPpgOYzwRDCzYQqHZ-jK

**Comment by PFRYCAC** [comment]
Thanks Erik!

One other bit maybe you will like to know is that our company name includes an ampersand, and this was automatically populated in the Publisher field which caused a parsing error preventing publishing.

**Comment by hougaard** [comment]
Yes, that's on my todo list, you're not the first with that.

**Comment by hougaard** [comment]
OK, instead of going to bed, I made some changes, version 9.0.0.324 is on it's way to AppSource. Try updating to that, the filter issue should be solved, and the ampersand in the publisher name should also work :)
(It typically takes 40 minutes for a new version to be available)

**Comment by PFRYCAC** [comment]
Thank you! And good night.
